### PR TITLE
[IMP] web, account: improve carets visibility

### PR DESCRIPTION
**Pull request by BastienFafchamps**
This PR improves various carets of the autocomplete dropdown and the list view sortable columns. It improves it's visibility and saves space when it overlaps with text.

Task: [4170216](https://www.odoo.com/odoo/project/133/tasks/4170216)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by clbr-odoo — APPROVED** [review]

File: addons/web/static/src/views/fields/many2one/many2one_field.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
                 <t t-if="hasExternalButton">
                     <button
                         type="button"
-                        class="btn btn-link text-action oi o_external_button"
+                        class="btn btn-link text-action oi o_external_button x-1"
```
Comment (pparidans) [inline addons/web/static/src/views/fields/many2one/many2one_field.xml:48]: `x-1` ? :thinking:

File: addons/web/static/src/views/list/list_renderer.scss [hunk 2]
```diff
@@ -89,12 +90,46 @@
...
                 text-align: right;
             }
+
+            .o_column_sortable {
+
+                .o_list_sortable_caret {
+                    background: var(--ListRenderer-thead-bg-color);
+                    position: absolute;
+                    right: .2rem;
+                    padding: 0 .2rem;
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.scss:111]: I'm I wrong, or there is a direct relation between the `.2rem` used here and the `min-width: .4rem` below?
If so, I would suggest extracting it into a variable and the corresponding calculation.

File: addons/web/static/src/views/list/list_renderer.scss [hunk 3]
```diff
@@ -89,12 +90,46 @@
...
+
+                &.table-active .o_list_header_label_spacer {
+                    width: 1.2rem;
+                    max-width: 1.2rem;
+                    min-width: .4rem;
+
+                    flex-shrink: 0;
+                    flex-grow: 1;
+                    flex-basis: 0;
+                }
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.scss:131]: I think there is some redundancy here between the `width`, `min-width`, `flex-basis`...

Here's my suggestion :
```suggestion
                .o_list_header_label_spacer {
                    flex: 0 0 0;
                }

                &.table-active .o_list_header_label_spacer {
                    flex: 1 0 0.4rem;
                    max-width: 1.2rem;
                }
```

In a nutshell, it means:
- in general, size of `0` and can neither shrink nor grow
- when the column is "active", base size is `0.4rem` and cannot shrink but is able to grow up to `1.2rem`

Note that there is a shorthand version for setting `flex-{grow,shrink,basis}` in one rule (cf. https://developer.mozilla.org/en-US/docs/Web/CSS/flex).

File: addons/web/static/src/views/list/list_renderer.xml [hunk 4]
```diff
@@ -28,12 +28,13 @@
                                 tabindex="-1">
                                 <t t-if="column.hasLabel and column.widget !== 'handle'">
                                 <t t-set="isNumeric" t-value="isNumericColumn(column)"/>
-                                    <div t-att-class="{'d-flex': true, 'flex-row-reverse': shouldReverseHeader(column)}"
+                                    <div class="d-flex"
                                         t-att-data-tooltip-template="isDebugMode ? 'web.FieldTooltip' : 'web.ListHeaderTooltip'"
                                         t-att-data-tooltip-info="makeTooltip(column)">
-                                        <span class="d-block min-w-0 text-truncate flex-grow-1" t-att-class="isNumeric ? 'o_list_number_th' : ''"
+                                        <div class="min-w-0 text-truncate flex-grow-1 flex-shrink-1" t-att-class="isNumeric ? 'o_list_number_th' : ''"
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.xml:34]: :warning: there is a tour's selector needing to be updated (cf. on runbot: failing `ir_model_fields_translation_en_tour` tour)

File: addons/web/static/src/views/list/list_renderer.scss (hunk 2 above)
Comment (BastienFafchamps) [reply addons/web/static/src/views/list/list_renderer.scss:111]: Not really, I just tried different sizes and chose what looked best

**Review by pparidans — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**